core/clustersmngr/clustersmngr.go

```diff
@@ -72,14 +74,18 @@ type clientsPool struct {
 	mutex   sync.Mutex
 }
 
-type ClusterClientConfig func(Cluster) *rest.Config
+type ClusterClientConfig func(Cluster) (*rest.Config, error)
 
 func ClientConfigWithUser(user *auth.UserPrincipal) ClusterClientConfig {
-	return func(cluster Cluster) *rest.Config {
+	return func(cluster Cluster) (*rest.Config, error) {
 		config := &rest.Config{
 			Host:            cluster.Server,
 			TLSClientConfig: cluster.TLSConfig,
 			Timeout:         kubeClientTimeout,
+			Dial: (&net.Dialer{
+				Timeout:   kubeClientDialTimeout,
+				KeepAlive: 15 * time.Second,
```

Comment:
Maybe a comment about 15s. Is it the default in client-go?

Reply:
Oh, this should be 30s, which is the default in client-go